BibBuddy/app/src/main/java/de/bibbuddy/NoteDao.java

```diff
@@ -252,6 +252,8 @@ public boolean linkNoteWithBook(Long bookId, Long noteId) {
    * @return returns the notes text value
    */
   public String findTextById(Long id) {
+    NoteTypeLut noteType = NoteTypeLut.TEXT;
```

Comment:
Currently this variable isn't used. Where in your code you check that the note is a text note?

Reply:
Claudia, this is the way to call the constructor. 

1. It is called when the value is from type 1 (TEXT). Here is a source, in which the enum constructors are explained:
[http://tutorials.jenkov.com/java/enums.html](url)
2. It is unnecessary to check if the NoteTypeLut.TEXT is equal to DatabaseHelper.NOTE_FILE_ID because the first one is from type Integer and the second one is from Type String. Moreover, when calling the method findTextById() the constructor is called with a value of 1.

I will not change it since the sharing function works properly.

Reply:
The noteType variable is not used at all so you should remove it or use it. 
Please consider the note type when you select notes because it is not considered yet. 
Instead of this code
`List<Long> notesList = noteDao.getAllNoteIdsForBook(book.getId());`  
you should write a new function where you get only the noteIds of the Text notes e.g. `getTextNoteIdsForBook(book.getId())`.  
Otherwise I will not approve your code like this.